x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/mapper/InferenceResultFieldMapper.java

```diff
@@ -168,8 +173,8 @@ private static void parseSingleField(DocumentParserContext context, MapperBuilde
 
         XContentParser parser = context.parser();
         String fieldName = parser.currentName();
-        Mapper mapper = context.getMapper(fieldName);
-        if (mapper == null || SemanticTextFieldMapper.CONTENT_TYPE.equals(mapper.typeName()) == false) {
+        MappedFieldType fieldType = context.mappingLookup().getFieldType(fieldName);
```

Comment:
Need `mappingLookup().getFieldType()` to search for dotted notation fields in multifields